niworkflows/viz/utils.py

```diff
@@ -24,10 +23,68 @@
 from niworkflows import NIWORKFLOWS_LOG
 from niworkflows.viz.validators import HTMLValidator
 
+try:
+    from shutil import which
+except ImportError:
+
+    def which(cmd):
+        """
+        A homemade which command
+
+        >>> from niworkflows.viz.utils import which
+        >>> which('ls')
+        True
+        >>> which('madeoutcommand')
+        False
+
+        """
+
+        try:
+            subprocess.run([cmd], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+        except OSError as e:
+            from errno import ENOENT
+            if e.errno == ENOENT:
+                return False
+            raise e
+        return True
+
 
 SVGNS = "http://www.w3.org/2000/svg"
 PY3 = version_info[0] > 2
 
+# Patch subprocess in python 2
+if version_info[0] < 3:
+    print('patching subprocess')
+    setattr(subprocess, 'DEVNULL', -3)
+
+    def _run(args, input=None, stdout=None, stderr=None, shell=False, check=False):
+        from collections import namedtuple
+
+        devnull = open(os.devnull, 'r+')
+        stdin = subprocess.PIPE if input is not None else None
+
+        if stdout == subprocess.DEVNULL:
+            stdout = devnull
+
+        if stderr == subprocess.DEVNULL:
+            stderr = devnull
+
+        proc = subprocess.Popen(args, stdout=stdout, shell=shell, stdin=stdin)
+        result = namedtuple('CompletedProcess', 'stdout stderr')
+
+        if not all((stdout == devnull, stderr == devnull)):
+            res = result(*proc.communicate(input=input))
+        else:
+            res = result(*proc.communicate(b''))
```

Comment:
I still don't understand why we don't want to pass input if `(stdout, stderr) == (devnull, devnull)` (I wasn't able to reproduce your error with any input I tried), but I don't think we have any situations where we have input and that case, so this should be fine.